### ISSUE TITLE
Boolean indexing with list gives wrong result

### ISSUE THREAD

**Issue by richardjgowers** [MDAnalysis/mdanalysis#729]
So currently
```python
ag = u.atoms[:3]

sl = [False, True, True]

ag[sl]

# returns len 3 AtomGroup with [0, 1, 1], instead of [1, 2]
```
This is because numpy interprets `[False, True, True]` to mean `[0, 1, 1]`, and then does integer indexing.  If however you slice with `np.array([False, True, True])` you get what is expected.  In the future however, this is changing so the list gets treated the same as a np array.

I'll change things to follow the [future numpy indexing rules](http://docs.scipy.org/doc/numpy-dev/release.html#indexing)

**Comment by kain88-de** [comment]
I'm not completely sure but I think numpy is getting around this by doing `np.asanyarray` object conversion for every input.

**Comment by richardjgowers** [comment]
What's the difference between `asarray` and `asanyarray`? Docs says asanyarray lets subclasses of array through, so should I be using that instead?

**Comment by kain88-de** [comment]
After reading the docs we almost always want a simple `asarray`. It would result in some surprises if we do some operations on a numpy.matrix instead of an array.

**Comment by richardjgowers** [comment]
Closed in #730
